Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ValidatorData.xml

```diff
@@ -2939,6 +2933,18 @@
         <CombineArray value="AP_IsVanillaStargate"/>
         <CombineArray value="AP_IsAPStargate"/>
     </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsVanillaStargateWarp">
+        <Value value="StargateWarp"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsAPStargateWarp">
+        <Value value="AP_StargateWarp"/>
+    </CValidatorUnitType>
+    <CValidatorCombine id="IsStargateWarp" parent="CValidatorCombine">
+        <!-- Used by Karax's Prodigal Engineer on Salvation -->
+        <!-- Override -->
+        <CombineArray value="AP_IsVanillaStargateWarp"/>
+        <CombineArray value="AP_IsAPStargateWarp"/>
+    </CValidatorCombine>

```

Comment:
See Robo comment

Reply:
Moved to ArchipelagoPlayerLotV and did a quick test on Salvation.